LLM.cs

```diff
@@ -0,0 +1,540 @@
+#endif
+        }
+
+        /// <summary>
+        /// Allows to set a LORA model to use in the LLM.
+        /// The model provided is copied to the Assets/StreamingAssets folder that allows it to also work in the build.
+        /// Models supported are in .bin format.
+        /// </summary>
+        /// <param name="path">path to LORA model to use (.bin format)</param>
+        public void SetLora(string path)
+        {
+            lora = SetModelLoraPath(path, true);
+#if UNITY_EDITOR
+            if (!EditorApplication.isPlaying) EditorUtility.SetDirty(this);
+#endif
+        }
+
+        /// <summary>
+        /// Set the chat template for the LLM.
+        /// </summary>
+        /// <param name="templateName">the chat template to use. The available templates can be found in the ChatTemplate.templates.Keys array </param>
+        public void SetTemplate(string templateName, bool setDirty = true)
+        {
+            chatTemplate = templateName;
+            if (started) llmlib?.LLM_SetTemplate(LLMObject, chatTemplate);
+#if UNITY_EDITOR
+            if (setDirty && !EditorApplication.isPlaying) EditorUtility.SetDirty(this);
+#endif
+        }
+
+        /// <summary>
+        /// Returns the chat template of the LLM.
+        /// </summary>
+        /// <returns>chat template of the LLM</returns>
+        public string GetTemplate()
+        {
+            return chatTemplate;
+        }
+
+        protected virtual string GetLlamaccpArguments()
+        {
+            // Start the LLM server in a cross-platform way
+            if (model == "")
+            {
+                LLMUnitySetup.LogError("No model file provided!");
+                return null;
+            }
+            string modelPath = GetModelLoraPath(model);
+            if (!File.Exists(modelPath))
+            {
+                LLMUnitySetup.LogError($"File {modelPath} not found!");
+                return null;
+            }
+            string loraPath = "";
+            if (lora != "")
+            {
+                loraPath = GetModelLoraPath(lora);
+                if (!File.Exists(loraPath))
+                {
+                    LLMUnitySetup.LogError($"File {loraPath} not found!");
+                    return null;
+                }
+            }
+
+            int numThreadsToUse = numThreads;
+            if (Application.platform == RuntimePlatform.Android && numThreads <= 0) numThreadsToUse = LLMUnitySetup.AndroidGetNumBigCores();
+
+            int slots = GetNumClients();
+            string arguments = $"-m \"{modelPath}\" -c {contextSize} -b {batchSize} --log-disable -np {slots}";
+            if (remote) arguments += $" --port {port} --host 0.0.0.0";
+            if (numThreadsToUse > 0) arguments += $" -t {numThreadsToUse}";
+            if (loraPath != "") arguments += $" --lora \"{loraPath}\"";
+            if (MMPROJmodel != "") arguments += $" --mmproj \"{MMPROJmodel}\"";

```

Comment:
Instead of copying a new LLM.cs file modify the Runtime/LLM.cs  to add the MMPROJmodel.
The MMPROJmodel needs to be treated similar to e.g. the loras, rather than providing this as a text, it needs some additional functionality to load it and make sure it is added inside the builds.
I will take over this part because it is quite involved.